docs/pr_checklist.md

```diff
@@ -0,0 +1,124 @@
+- Other notes TBD
+    - Core is a lot more subjective given the breadth of posted changes
+
+## Keyboard PRs
+
+Closed PRs (for inspiration, previous sets of review comments will help you eliminate ping-pong of your own reviews):
+https://github.com/qmk/qmk_firmware/pulls?q=is%3Apr+is%3Aclosed+label%3Akeyboard
+
+- `info.json`
+    - valid URL
+    - valid maintainer
+    - displays correctly in Configurator (press Ctrl+Shift+I to preview local file, turn on fast input to verify ordering)
+- `readme.md`
+    - standard template should be present
+    - flash command has `:flash` at end
+    - valid hardware availability link (unless handwired) -- private groupbuys are okay, but one-off prototypes will be questioned. If open-source, a link to files should be provided.
+- `rules.mk`
+    - removed `MIDI_ENABLE`, `FAUXCLICKY_ENABLE` and `HD44780_ENABLE`
+    - modified `# Enable Bluetooth with the Adafruit EZ-Key HID` -> `# Enable Bluetooth`
+    - No `(-/+size)` comments related to enabling features
+    - Remove the list of alternate bootloaders if one has been specified
+    - No re-definitions of the default MCU parameters if same value, when compared to the equivalent MCU in [mcu_selection.mk](https://github.com/qmk/qmk_firmware/blob/master/quantum/mcu_selection.mk)
+- keyboard `config.h`
+    - don't repeat `MANUFACTURER` in the `PRODUCT` value
+    - no `#define DESCRIPTION`
+    - no Magic Key Options, MIDI Options or HD44780 configuration
+    - user preference configurable `#define`s need to be moved to keymap `config.h`
+    - "`DEBOUNCE`" instead of "`DEBOUNCING_DELAY`"
+    - bare minimum required code for a board to boot into QMK should be present
+        - initialisation code for the matrix and critical devices
+        - mirroring existing functionality of a commercial board (like custom keycodes and special animations etc.) should be handled through non-`default` keymaps
+- `keyboard.c`
+    - empty `xxxx_xxxx_kb()` or other weak-defined default implemented functions removed
+    - commented-out functions removed too
+    - `matrix_init_board()` etc. migrated to `keyboard_pre_init_kb()`, see: [keyboard_pre_init*](https://docs.qmk.fm/#/custom_quantum_functions?id=keyboard_pre_init_-function-documentation)
+    - prefer `CUSTOM_MATRIX = lite` if custom matrix used, allows for standard debounce, see [custom matrix 'lite'](https://docs.qmk.fm/#/custom_matrix?id=lite)
+- `keyboard.h`
+    - `#include "quantum.h"` appears at the top
+    - `LAYOUT` macros should use standard definitions if applicable
+        - Use the Community Layout macro names where they apply (preferred above `LAYOUT`/`LAYOUT_all`)
+- keymap `config.h`
+    - no duplication of `rules.mk` or `config.h` from keyboard
+- `keymaps/default/keymap.c`
+    - `QMKBEST`/`QMKURL` removed (sheesh)
+    - lower/raise should prefer (instead of massive switch statement with layer_on/off) if no "direct to adjust" keycode:
```

Comment:
Can you clarify this a bit more? Didn't quite get it after a few read throughs.

Reply:
This bit is saying to prefer this:

```
layer_state_t layer_state_set_user(layer_state_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

over this:

```
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
    case LOWER:
      if (record->event.pressed) {
        layer_on(_LOWER);
        update_tri_layer(_LOWER, _RAISE, _ADJUST);
      } else {
        layer_off(_LOWER);
        update_tri_layer(_LOWER, _RAISE, _ADJUST);
      }
      return false;
      break;
    case RAISE:
      if (record->event.pressed) {
        layer_on(_RAISE);
        update_tri_layer(_LOWER, _RAISE, _ADJUST);
      } else {
        layer_off(_RAISE);
        update_tri_layer(_LOWER, _RAISE, _OTHER);
      }
      return false;
      break;
  }
  return true;
};
```

unless there's an `ADJUST` case -- since the `update_tri_state_layer` function can't coexist with a `MO(_ADJUST)`.

Reply:
Reworded in latest commit.